### PR TITLE
feat(sunburst): add innerRadiusRatio, renderRootNode prop to control size of inner circle

### PR DESCRIPTION
**Pull request by adamberro**
* Adds a prop similar to the pie chart's `innerRadius` prop to change the size of the sunburst chart's inner circle.
* Adds a prop to render the root node in the center of the chart.

Fixes https://github.com/plouc/nivo/issues/1649

Please let me know if this approach works or if it would be more desirable to give more flexibility such as the option to configure the type of d3 scale, or if there are any other issues or anything else is missing.

### CODE REVIEW

**Comment by adamberro** [comment]
Hi @plouc, anything we can do to get the review process started? Thank you!

**Comment by vicaub** [comment]
I really like this feature and need it on my project too!
But probably it would look better if all layer had the same radius length

File: packages/sunburst/src/hooks.ts [hunk 1]
```diff
@@ -68,6 +75,12 @@ export const useSunburst = <RawDatum>({
         // are going to be computed first
         const sortedNodes = sortBy(descendants, 'depth')
 
+        const innerRadiusOffset = radius * Math.min(innerRadiusRatio, 1)
+
+        const maxDepth = Math.max(...sortedNodes.map(n => n.depth))
+
+        const scale = d3ScaleRadial().domain([0, maxDepth]).range([innerRadiusOffset, radius])
```
Comment (vicaub) [inline packages/sunburst/src/hooks.ts:82]: using d3 scaleLinear instead of scaleRadial would render better in my opinion

**Comment by adamberro** [comment]
Hi @plouc, just checking in on whether we might be able to merge this. I'm happy to make any changes as needed. Thanks!

File: packages/arcs/src/centers.ts [hunk 2]
```diff
@@ -45,6 +45,11 @@ export const useArcCentersTransition = <Datum extends DatumWithArc, ExtraProps =
     mode: ArcTransitionMode = 'innerRadius',
     extra?: TransitionExtra<Datum, ExtraProps>
 ) => {
+    // center label of root node
```
Comment (adamberro) [inline packages/arcs/src/centers.ts:48]: This centers the root most label. Not sure if there is a more proper way to handle this, but I don't think it was accounted for originally

Screen Shot 2021-11-26 at 9 09 56 AM
.

File: packages/sunburst/src/hooks.ts (hunk 1 above)
Comment (adamberro) [reply packages/sunburst/src/hooks.ts:82]: scaleLinear looks quite different from the original implementation: 
Screen Shot 2021-11-26 at 9 57 09 AM

Maybe we could update the sunburst props to include setters for the arc inner/outer radii. I'm not sure if this reaches a happy medium between customizability and ease of use, but my initial thought for the api would be something like: 
```
type GetInnerRadius = (currentNode: HierarchyRectangularNode<RawDatum>, nodes: HierarchyRectangularNode<RawDatum>[]) => number
```
@plouc Thoughts on this approach?

**Comment by plouc** [comment]
@adamberro, sorry for the late reply, I'll try to run this branch locally and review the PR in the upcoming weeks.

**Comment by lers92** [comment]
Also looking forward to this new feature 😄

**Comment by tkonopka** [comment]
bump

**Comment by rnsv** [comment]
bump

**Comment by TotomiEcio** [comment]
Hi! I hope this MR is able to merge. I would like this changes. Thank you

**Comment by plouc** [comment]
Superseded by https://github.com/plouc/nivo/pull/2696.

**Pull request closed without merging**